### ISSUE TITLE
Namespacing issues

### ISSUE THREAD

**Issue by stevschmid** [komposable/komponent#117]
Hey guys

I just submitted the PR #116 but I stumbled upon a problem with it: I assumed (wrongly) that a component generated via `rails g component 'hello/world'` would create the main view as `components/hello/world/_world.html.slim`. Instead, it creates `components/hello/world/_hello_world.html.slim`

is there a rationale behind this? And why is the generated ruby component not properly namespaced as well?
Expected: `module Hello::WorldComponent`
Actual: `module HelloWorldComponent`

Is this an oversight or deliberate behavior? Because if I execute `rails g component 'foo/bar'` and `rails g component 'foo_bar` we end up with two modules with the same name.

**Comment by Spone** [comment]
Hi @stevschmid thanks for opening this issue and for your PR :)

The idea behind the `components/hello/world/_hello_world.html.slim` naming is to keep the filename explicit enough about the current component namespace.

If you have several components named `item` for instance, each in their own namespace, it quickly becomes a mess in your editor when you open several of them in tabs, because they would all be named `_item.html.slim`

That's why we chose to keep the whole namespace in the component view filename.

About the Ruby module name: we sticked with `HelloWorldComponent` because we did not want to deal with the added complexity of relying on Ruby namespaces (you can create collisions with existing classes pretty easily on large projects).

Does this answer your questions?

**Comment by Spone** [comment]
I agree those are valid issues to consider. We'll discuss it and let you know (probably second half of September, I'm going to be on vacation until then).

**Comment by stevschmid** [comment]
No worries. Thanks for getting back to me so quickly. Enjoy your vacation!

**Comment by stevschmid** [comment]
Hey @Spone 

Thanks for getting back to me. Sounds great!
